% LaTeX file for Chapter 04
<<'preamble04',include=FALSE>>=
library(knitr)
opts_chunk$set(
    fig.path='figure/ch04_fig', 
    self.contained=FALSE,
    cache=FALSE
) 
@

\chapter{Discussion}

\section{Conclusion} 
In this thesis we investigated, how the relative abundance of spliced and unspliced reads differs between experimental conditions. Changes to these relative abundances are directly linked to gene regulation and methods that are capable of detecting these differences already exist (e.g. \emph{BRIE2} and \emph{eisaR}). We identified two main sources of mapping uncertainty: reads mapping to multiple genes, and reads mapping to both spliced and unspliced versions of a gene. We proposed two approaches to deal with these: i) \emph{DEXSeq} on USA estimated counts, which accounts for the first, and \emph{DifferentialRegulation} on the USA-based equivalence classes, which also uses a latent variable model for reads mapping to multiple genes, hence accounting for both sources of mapping uncertainty. We investigated the performance of all methods on two semi-simulated data sets. The semi-simulated data sets were created from real scRNA-seq data with four biological replicates. In a first step, we introduced an arbitrary differential effect to a subset of genes and cells by inverting the counts of spliced and unspliced reads. In one of the two simulations we also added DGE to a second subset of genes and cells as an additional nuisance parameter. Multi-mapping uncertainty was considered next to make the semi-simulated data sets more realistic. This was achieved by simulating, at the read level, with \emph{minnow} and subsequently aligning simulated reads with \emph{alevin-fry}, where we used the two semi-simulated data sets as input for \emph{minnow}. We then analysed the performance of \emph{BRIE2}, \emph{eisaR}, \emph{DEXSeq} and \emph{DifferentialRegulation} in detecting the differential genes by comparing ROC and TPR v. FDR curves. From this analysis we found that \emph{DifferentialRegulation} controls the FDR well for both data sets in comparison to the other three methods. Additionally, we examined the results stratified by gene abundance levels to investigate how robust the methods are to gene abundance with and without DGE. From this analysis we concluded that \emph{DifferentialRegulation} is the only method with good TPR and well calibrated FDR across all levels of gene abundance. Next, we did a null analysis on the original data set where we compared the distribution of p-values for all three possible group separations. We found that \emph{BRIE2} had inflated p-values for all three group separations, whereas \emph{DifferentialRegulation} had no inflated p-values at all. From the null analysis (where no differences between groups are expected) it was also shown that \emph{DifferentialRegulation} seems to be marginally conservative as there was a tendency for inflation towards one. Ultimately, we ran a computational benchmark on the null data where we applied all four methods on all three possible group separations and averaged the runtime. From the computational benchmark it was shown that the better performance of \emph{DifferentialRegulation} comes with a longer runtime, however, there is also the possibility to run the method without equivalence classes, which is slightly less accurate however much faster. From the previous analyses, we show that \emph{DifferentialRegulation} had overall better results than the other methods, however, there are also some caveats to the method, for example, \emph{DifferentialRegulation} cannot deal with additional covariates e.g. batch effects, whereas \emph{BRIE2} and \emph{DEXSeq} can.

\section{Future directions}
\emph{DifferentialRegulation} has already been published on the Bioconductor project, which is an open source software that promotes reproducible analysis of data from emerging biological assays. Further, it is planned to extend the use case of \emph{DifferentialRegulation} from only scRNA-seq data to bulk RNA-seq data, which allows to perform differential analyses at the transcript level. This will enable a double analysis framework: i) identify cell-type specific changes from scRNA-seq data, but at the gene-level due to low transcript resolution, and ii) discover individual differentially regulated transcripts from bulk RNA-seq data, although from an aggregation of cell types. Ultimately, the work of this thesis is part of a future paper which needs some additional analyses on different data sets and some changes to the simulation algorithm.